components/results_table/commons/results_table.lua

```diff
@@ -101,8 +109,203 @@ function BaseResultsTable:getQueryType()
+
+	local conditions = self:buildBaseConditions()
+
+	if self.args.additionalConditions then
+		return conditions .. self.args.additionalConditions
+	end
+
+	return conditions
+end
+
+function BaseResultsTable:buildBaseConditions()
+	local args = self.args
+
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{self:buildOpponentConditions()}
+
+	if args.game then
+		conditions:add{ConditionNode(ColumnName('game'), Comparator.eq, args.game)}
+	end
+
+	local startDate = args.startdate or args.sdate
+	if startDate then
+		-- intentional > here to keep it as is in current modules
+		-- possibly change to >= later
+		conditions:add{ConditionNode(ColumnName('date'), Comparator.gt, startDate)}
+	end
+
+	local endDate = args.enddate or args.edate
+	if endDate then
+		-- intentional < here to keep it as is in current modules
+		-- possibly change to <= later
+		conditions:add{ConditionNode(ColumnName('date'), Comparator.lt, endDate)}
+	end
+
+	if args.placement then
+		conditions:add{ConditionNode(ColumnName('placement'), Comparator.eq, args.placement)}
+	elseif not Logic.readBool(args.ignorePlacement) then
+		conditions:add{ConditionNode(ColumnName('placement'), Comparator.neq, '')}
+	end
+
+	if args.tier then
+		local tierConditions = ConditionTree(BooleanOperator.any)
+		for _, tier in pairs(mw.text.split(args.tier, ',%s?')) do
+			tierConditions:add{ConditionNode(ColumnName('liquipediatier'), Comparator.eq, tier)}
+		end
+		conditions:add{tierConditions}
+	end
+
+	return conditions:toString()
+end
+
+-- todo: adjust once #1802 is done
+function BaseResultsTable:buildOpponentConditions()
+	local config = self.config
+	local opponent = config.opponent
+
+	local opponentConditions = ConditionTree(BooleanOperator.any)
+
+	if config.queryType == SOLO_TYPE or config.queryType == COACH_TYPE then
+		opponent = config.resolveOpponent
+			and mw.ext.TeamLiquidIntegration.resolve_redirect(opponent)
+			or opponent
+
+		local opponentWithUnderscore = opponent:gsub(' ', '_')
+
+		local prefix
+		if config.queryType == SOLO_TYPE then
+			prefix = config.playerPrefix
+			opponentConditions:add{
+				ConditionTree(BooleanOperator.all):add{
+					ConditionNode(ColumnName('opponenttype'), Comparator.eq, Opponent.solo),
+					ConditionNode(ColumnName('opponentname'), Comparator.eq, opponent),
+				},
+				ConditionTree(BooleanOperator.all):add{
+					ConditionNode(ColumnName('opponenttype'), Comparator.eq, Opponent.solo),
+					ConditionNode(ColumnName('opponentname'), Comparator.eq, opponentWithUnderscore),
+				},
+			}
+		else
+			prefix = config.coachPrefix
+		end
+
+		for playerIndex = 1, config.playerLimit do
+			opponentConditions:add{
+				ConditionNode(ColumnName('opponentplayers_' .. prefix .. playerIndex), Comparator.eq, opponent),
+				ConditionNode(ColumnName('opponentplayers_' .. prefix .. playerIndex), Comparator.eq, opponentWithUnderscore),
+			}
+		end
+
+		return opponentConditions
+	end
+
+	opponent = Team.queryDB('teampage', opponent)
```

Comment:
If queryType == TEAM_TYPE then

Reply:
due to `if config.queryType == SOLO_TYPE or config.queryType == COACH_TYPE then` before it and that returning we are impicitly in the case

Reply:
done